src/ast/ast_build_filter_tree.c

```diff
@@ -134,6 +134,23 @@ static FT_FilterNode *_convertIntegerOperator(const cypher_astnode_t *expr) {
 	return FilterTree_CreateExpressionFilter(exp);
 }
 
+static FT_FilterNode *_convertLabelsOperator(const cypher_astnode_t *expr) {
```

Comment:
I think it would be much more efficient if we would introduce a new arithmetic function (`has_label`)
which given a node `n` object and a label id `L` checks to see if label_matrix_L[ID(n), ID(n)] is set.